File: code/platform/src/main/java/org/fbme/ide/platform/debugger/DeviceConnection.java
package org.fbme.ide.platform.debugger;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.mps.openapi.model.SNode;

import java.io.Closeable;
import java.io.IOException;

public interface DeviceConnection extends Closeable {

    void deployResource(SNode resource) throws IOException;

    void addWatch(@NotNull Watchable watchable) throws IOException;

    void removeWatch(@NotNull Watchable watchable) throws IOException;

    String readWatches() throws IOException;

    boolean isAlive();
}

File: code/platform/src/main/kotlin/org/fbme/ide/platform/debugger/DeviceConnection.kt
package org.fbme.ide.platform.debugger

import org.fbme.lib.iec61499.declarations.ResourceDeclaration
import java.io.Closeable
import java.io.IOException

interface DeviceConnection : Closeable {
    val isAlive: Boolean

    @Throws(IOException::class)
    fun deployResource(resource: ResourceDeclaration)

    @Throws(IOException::class)
    fun killResource(resource: ResourceDeclaration)

    @Throws(IOException::class)
    fun deleteResource(resource: ResourceDeclaration)

    @Throws(IOException::class)
    fun createResourceNetwork(resource: ResourceDeclaration)

    @Throws(IOException::class)
    fun startResource(resource: ResourceDeclaration)

    @Throws(IOException::class)
    fun addWatch(watchable: Watchable)

    @Throws(IOException::class)
    fun removeWatch(watchable: Watchable)

    @Throws(IOException::class)
    fun readWatches(): String
}